### PR TITLE
[SPARK-26564] Fix wrong assertions and error messages for parameter checking

### PR DESCRIPTION
**Pull request by sekikn**
## What changes were proposed in this pull request?

If users set equivalent values to spark.network.timeout and spark.executor.heartbeatInterval, they get the following message:

```
java.lang.IllegalArgumentException: requirement failed: The value of spark.network.timeout=120s must be no less than the value of spark.executor.heartbeatInterval=120s.
```

But it's misleading since it can be read as they could be equal. So this PR replaces "no less than" with "greater than". Also, it fixes similar inconsistencies found in MLlib and SQL components.

## How was this patch tested?

Ran Spark with equivalent values for them manually and confirmed that the revised message was displayed.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
It's okay but mind taking a look for similar instances? I think there are more.

**Comment by sekikn** [comment]
Thank you for the comment @HyukjinKwon. As you said, I found two additional mistakes by grep'ing the codebase.

```
./mllib/src/main/scala/org/apache/spark/ml/optim/WeightedLeastSquares.scala:  require(maxIter >= 0, s"maxIter must be a positive integer: $maxIter")
```

```
./sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala:      require(capacity < 512000000, "Cannot broadcast more than 512 millions rows")
```

But for the first one, I can't decide which of the constraint and the message is correct since I'm not so familar with ML. It doesn't seem to make sense to execute it with maxIter=0, but is there a use-case for that, such as a dry-run...?

**Comment by HyukjinKwon** [comment]
I think it's okay to leave them as are. Let's just get the current change in then.

**Comment by HyukjinKwon** [comment]
ok to test

**Review by HyukjinKwon — APPROVED** [review]

**Comment by srowen** [comment]
In both of the additional cases above, the assertion is wrong and the message is right. You are welcome to fix both of them here, but I'm OK merging this too.

**Comment by sekikn** [comment]
Thank you for the advice @HyukjinKwon and @srowen. I'll fix them within this PR so let me update it.

**Review by srowen — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala [hunk 1]
```diff
@@ -413,7 +413,7 @@ private[execution] final class LongToUnsafeRowMap(val mm: TaskMemoryManager, cap
 
   private def init(): Unit = {
     if (mm != null) {
-      require(capacity < 512000000, "Cannot broadcast more than 512 millions rows")
+      require(capacity <= 512000000, "Cannot broadcast more than 512 millions rows")
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala]: I think the previous condition was true.

https://github.com/apache/spark/blob/79c66894296840cc4a5bf6c8718ecfd2b08bcca8/sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala#L80-L83

negated `numRows >= 512000000` should be `capacity < 512000000`

Comment (srowen) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala]: Ah, good catch. Sorry @sekikn could we change this one more time, to use `<` and then fix the error message instead?

Comment (sekikn) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala]: Sure! I'm going to update the PR.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**